File: r/inst/examples/html_extraction.R
# html_extraction — build a PDF from Markdown, open it, render the whole
# document to HTML. stop() (non-zero exit) on any assertion failure (CI smoke).
library(pdfoxide)

md <- "# Hello pdf_oxide\n\nThis is a **R** regression example.\n"

pdf <- pdf_from_markdown(md)
doc <- pdf_open_from_bytes(pdf_to_bytes(pdf))

html <- pdf_to_html_all(doc)
cat("--- html (all) ---\n", html, "\n", sep = "")

if (!grepl("<", html, fixed = TRUE)) stop("ASSERT FAILED: html does not contain '<'")
if (!grepl("pdf_oxide", html, fixed = TRUE)) {
  stop("ASSERT FAILED: html does not contain 'pdf_oxide'")
}

pdf_close(doc)
cat("HTML OK\n")

File: dart/example/html_extraction.dart
// html_extraction — build a PDF from Markdown, open it, render HTML.
//
// Shared regression scenario (mirrored across language bindings). Exits
// non-zero on any failed assertion; prints "HTML OK" on success.
import 'dart:io';

import 'package:pdf_oxide/pdf_oxide.dart';

void main() {
  final pdf = Pdf.fromMarkdown(
      '# Hello pdf_oxide\n\nThis is a **Dart** regression example.\n');
  final doc = PdfDocument.openFromBytes(pdf.toBytes());
  try {
    final html = doc.toHtmlAll();
    print('--- html (all) ---');
    print(html);

    if (!html.contains('<')) {
      stderr.writeln("assertion failed: html does not contain '<'");
      exit(1);
    }
    if (!html.contains('pdf_oxide')) {
      stderr.writeln("assertion failed: html does not contain 'pdf_oxide'");
      exit(1);
    }

    print('HTML OK');
  } finally {
    doc.close();
    pdf.close();
  }
}
